firewood/src/merkle/node.rs

```diff
@@ -451,87 +444,83 @@ impl Storable for Node {
     fn serialized_len(&self) -> u64 {
         Meta::SIZE as u64
             + match &self.inner {
-                NodeType::Branch(n) => {
-                    // TODO: add path
-                    n.serialized_len()
-                }
+                NodeType::Branch(n) => n.serialized_len(),
                 NodeType::Extension(n) => n.serialized_len(),
                 NodeType::Leaf(n) => n.serialized_len(),
             }
     }
 
     fn serialize(&self, to: &mut [u8]) -> Result<(), ShaleError> {
         trace!("[{self:p}] Serializing node");
-        let mut cur = Cursor::new(to);
-
-        let mut attrs = match self.root_hash.get() {
-            Some(h) => {
-                cur.write_all(&h.0)?;
-                NodeAttributes::ROOT_HASH_VALID
-            }
-            None => {
-                cur.write_all(&[0; 32])?;
-                NodeAttributes::empty()
-            }
+        let mut cursor = Cursor::new(to);
+
+        let root_hash = self.root_hash.get().map(|hash| hash.0);
```

Comment:
I can understand this probably because I wrote the first version. To better the understanding, I do agree to separate the `root_hash` and `encoded` logic respectively and leave block comments may also help. Also, with the simplifying of NodeAttributes it may be easier to read as well.